io/EptAddonWriter.cpp

```diff
@@ -0,0 +1,258 @@
+/******************************************************************************
+* Copyright (c) 2019, Connor Manning (connor@hobu.co)
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include "EptAddonWriter.hpp"
+
+#include <cassert>
+
+#include <json/json.h>
+
+#include <arbiter/arbiter.hpp>
+
+#include "private/EptSupport.hpp"
+
+namespace pdal
+{
+
+namespace
+{
+    const StaticPluginInfo s_info
+    {
+        "writers.ept_addon",
+        "EPT Writer",
+        "http://pdal.io/stages/writers.ept.html",
+        { "ept_addon", "ept-addon" }
+    };
+
+    std::string getTypeString(Dimension::Type t)
+    {
+        const auto base(Dimension::base(t));
+        if (base == Dimension::BaseType::Signed) return "signed";
+        if (base == Dimension::BaseType::Unsigned) return "unsigned";
+        if (base == Dimension::BaseType::Floating) return "float";
+        throw std::runtime_error("Invalid dimension type");
```

Comment:
Boo on throwing this.  Either throwError(), or throw something that is caught by caller and then throwError().  If someone gets this, they'll have no clue without a traceback.

Reply:
Fixed in 514524e.